### ISSUE TITLE
does not build on 4.04+beta due to warning-as-errors setting

### ISSUE THREAD

**Issue by gasche** [andersfugmann/amqp-client#2]
No version of amqp-client builds on 4.04+beta due to unused module warnings.

opam-builder report: <http://opam.ocamlpro.com/builder/html/amqp-client/amqp-client.1.0.0/9eb8b9a622cf60ffdc2715a47809315f>

```
File "amqp_queue.ml", line 2, characters 0-35:
Warning 60: unused module Connection.
File "amqp_exchange.ml", line 2, characters 0-35:
Warning 60: unused module Connection.
File "amqp_queue.ml", line 1:
Error: Some fatal warnings were triggered (1 occurrences)
File "amqp_exchange.ml", line 1:
Error: Some fatal warnings were triggered (1 occurrences)
```

**Comment by andersfugmann** [comment]
Thanks for reporting. I have made a bugfix release and submitted a patch for opam-repository.

**Comment by gasche** [comment]
Thanks a lot. Does your opam-repository include setting an upper bound on the non-building release?

**Comment by andersfugmann** [comment]
What do you mean by upper bound? 
Amqp-client (>= 1.0.0) should compile with all versions of ocaml  >= 4.0.3.

**Comment by gasche** [comment]
I mean that right now 1.0.0 tries to install under 4.04 and fails at build
time. Opam packages should never fail at build time, they should not be
installable if they would fail, so 1.0.0 needs to add (ocaml-version <
"4.04") in its constraints.

**Comment by andersfugmann** [comment]
Ahh. I see. 
Do you know what the recommended practice with opam-repository is? Should I update the opam file for the existing 1.0.0. release or just delete 1.0.0 since 1.0.1 supersedes it 

(which is pending since omake cannot compile with 4.0.4-beta it seems)

**Comment by gasche** [comment]
Yes, you should update the opam file for 1.0.0. I've been doing that in spades for other packages recently. [opam-builder](http://opam.ocamlpro.com/builder/html/report-full.html) is a super-useful tool to get a global view of the package ecosystem, and we try to fix every "Fail" package and understand compatibility issues caused by 4.04.

omake reports as building on [the opam-builder report](http://opam.ocamlpro.com/builder/html/report-full.html). Does building omake fail on your machine, or does omake fails to build your project? Do you have an error message?

**Comment by andersfugmann** [comment]
Ok, Thanks. I will update the opam file for 1.0.0.



Wrt. omake, it seems to be travis who cannot compile omake, which means that build checks fails on the PR agains opam-respository to add version 1.0.1 of amqp-client. Below is a log excerpt.
```
[ERROR] The compilation of omake failed at "make all
        PREFIX=/home/travis/.opam/4.04.0+beta1".
#=== ERROR while installing omake.0.9.8.6-0.rc1 ===============================#
# opam-version 1.2.2
# os           linux
# command      make all PREFIX=/home/travis/.opam/4.04.0+beta1
# path         /home/travis/.opam/4.04.0+beta1/build/omake.0.9.8.6-0.rc1
# compiler     4.04.0+beta1
# exit-code    2
# env-file     /home/travis/.opam/4.04.0+beta1/build/omake.0.9.8.6-0.rc1/omake-14285-d8393c.env
# stdout-file  /home/travis/.opam/4.04.0+beta1/build/omake.0.9.8.6-0.rc1/omake-14285-d8393c.out
# stderr-file  /home/travis/.opam/4.04.0+beta1/build/omake.0.9.8.6-0.rc1/omake-14285-d8393c.err
### stdout ###
# [...]
# --- Checking if ocamldep understands -modules... (yes)
# --- Checking for hevea... (FAILED - no hevea found)
# --- Checking for date... (found /bin/date)
# --- Checking for date... (found /bin/date)
# *** omake: finished reading OMakefiles (1.77 sec)
# *** omake: saved .omakedb
# - build src/main omake.opt
# + ocamlopt.opt -warn-error A -w a -I . -I ../libmojave -I ../util -I ../magic -I ../ast -I ../ir -I ../env -I ../exec -I ../eval -I ../shell -I ../build -I ../builtin -o omake.opt unix.cmxa ../libmojave/lm.cmxa ../util/util.cmxa ../magic/magic.cmxa ../ast/ast.cmxa ../ir/ir.cmxa ../env/env.cmxa ../exec/exec.cmxa ../eval/eval.cmxa ../shell/shell.cmxa ../build/build.cmxa ../builtin/builtin.cmxa omake_shell.cmx omake_main.cmx ../clib/clib.a -cclib -lncurses -cclib -lreadline
# *** omake: 1198/1203 targets are up to date
# *** omake: failed (25.46 sec, 234/234 scans, 342/504 rules, 786/2363 digests)
### stderr ###
# [...]
# ../clib/clib.a(lm_terminfo.o): In function `caml_tgetstr':
# /home/travis/.opam/4.04.0+beta1/build/omake.0.9.8.6-0.rc1/src/clib/lm_terminfo.c:72: undefined reference to `tigetstr'
# ../clib/clib.a(lm_terminfo.o): In function `load_terminfo':
# /home/travis/.opam/404.0+beta1/build/omake.0.9.8.6-0.rc1/src/clib/lm_terminfo.c:35: undefined reference to `setupterm'
# collect2: error: ld returned 1 exit status
# File "caml_startup", line 1:
# Error: Error during linking
# *** omake: targets were not rebuilt because of errors:
#    src/main/omake.opt
# make: *** [all] Error 2
```
(See the full log here) https://travis-ci.org/ocaml/opam-repository/jobs/155560601)

**Comment by gasche** [comment]
This looks like a missing `ncurses` dependency revealed recently by ocaml/opam-repository#6657. I'll bring the issue upstream (against opam-repo), thanks.

**Comment by gasche** [comment]
So looking at the sources of omake, it looks like it's not actually related
to this conf-ncurses change, it's a trickier omake-specific issue related
to ncurses/terminfo. we should report anyway.

**Comment by gasche** [comment]
(After investigating the issue more, the omake build failure is due to an ocaml compiler issue caused by ocaml/ocaml#464 and fixed in ocaml/ocaml#758, so it is not omake-specific and should be fixed in the 4.04 branch and in the upcoming second beta release.)

**Comment by andersfugmann** [comment]
It seems the prolems have been resolved. Thanks Gabriel for reporting and fixing.
